### PR TITLE
GUI: Numerical columns can be sorted numerically

### LINKED ISSUE #1419: GUI: Download sizes should be sorted as numbers, not strings.
Steps to reproduce:

1. Open the GUI
1. Sort by download size.

### PR DESCRIPTION
**Pull request by Postremus**
Moved the creation of the 3 different sorter kinds into different methods for:
string based, integer based and checkbox based
fixes #1419

### CODE REVIEW

**Comment by Keith-UCT** [comment]
Thank you for doing this. Will be very helpful when uninstalling and installing mods.

File: GUI/MainModList.cs [hunk 1]
```diff
@@ -18,38 +18,71 @@ private void UpdateFilters(Main control)
...
-                        row => row.Cells[this.m_Configuration.SortByColumnIndex].Value.ToString());
+                return Sort(rows, CreateIntegerSorter());
             }
-            // Update the column sort glyph
-            this.ModList.Columns[this.m_Configuration.SortByColumnIndex].HeaderCell.SortGlyphDirection =
-                this.m_Configuration.SortDescending ? SortOrder.Descending : SortOrder.Ascending;
+            return Sort(rows, CreateDefaultSorter());
+        }
+
+        private IEnumerable<DataGridViewRow> Sort<T>(IEnumerable<DataGridViewRow> rows, Func<DataGridViewRow, T> sortFunction)
```
Comment (pjf) [inline GUI/MainModList.cs]: Sweet! Today I learnt the `Func<>` keyword in C#. :)

Comment (plague006) [inline GUI/MainModList.cs]: https://youtu.be/XxaP19pY1FI?t=7m30s

Comment (pjf) [inline GUI/MainModList.cs]: @plague006 : I have *no idea* what I just watched, but it was *amazing*.

**Comment by pjf** [comment]
Merged with thanks, and a small adjustment to use methods rather than closures (cd47a9b)

File: GUI/MainModList.cs (hunk 1 above)
Comment (plague006) [inline GUI/MainModList.cs]: You learned about the func. :smile:

**Pull request merged**